src/connection_manager.rs

```diff
@@ -107,16 +108,23 @@ impl ConnectionManager {
 
         // Send message to all Elders concurrently
         let mut tasks = Vec::default();
-        for elder in &self.elders {
+
+        // clone elders as we want to update them in this process
+        for socket in self.elders.clone().keys() {
```

Comment:
Wouldn't it be better to clone the keys only rather than the treemap? e.g. `self.elders.keys().cloned().collect();`